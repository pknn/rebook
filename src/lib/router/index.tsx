import React from 'react';
import { createBrowserHistory } from 'history';
import {
  connectRouter,
  routerMiddleware as getRouterMiddleware,
  ConnectedRouter as ConnectedRouterComponent
} from 'connected-react-router';

export const browserHistory = createBrowserHistory();
export const routerReducer = connectRouter(browserHistory);
export const routerMiddleware = getRouterMiddleware(browserHistory);
export const ConnectedRouter: React.FC = ({ children }) => (
  <ConnectedRouterComponent history={browserHistory}>
    {children}
  </ConnectedRouterComponent>
);
