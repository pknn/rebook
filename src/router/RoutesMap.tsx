import React from 'react';
import { Switch, Route } from 'react-router-dom';

import HomePage from '../pages/Home';
import NotFoundPage from '../pages/NotFound';

const RoutesMap: React.FunctionComponent = () => (
  <Switch>
    <Route exact path="/">
      <HomePage />
    </Route>
    <Route path="*">
      <NotFoundPage />
    </Route>
  </Switch>
);

export default RoutesMap;
