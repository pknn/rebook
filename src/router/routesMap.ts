import HomePage from '../pages/Home';
import NotFoundPage from '../pages/NotFound';

export interface Route {
  path: string;
  title: string;
  exact?: boolean;
  component: React.FunctionComponent;
}

const routesMap: Route[] = [
  {
    path: '/',
    title: 'Home',
    exact: true,
    component: HomePage
  },
  {
    path: '/not-found',
    title: 'Not Found',
    component: NotFoundPage
  }
];

export default routesMap;
