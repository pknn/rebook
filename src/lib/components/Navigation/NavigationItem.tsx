import React from 'react';
import { Link } from 'react-router-dom';
import { Route } from 'router/routesMap';

interface NavigationItemProps {
  route: Route;
  isCurrent: boolean;
}

const NavigationItem: React.FC<NavigationItemProps> = ({
  route,
  isCurrent
}) => (
  <li
    className={`mx-2 text-base ${
      isCurrent ? 'text-yellow-500' : 'text-gray-600 hover:text-yellow-500'
    }`}
  >
    <Link to={route.path}>{route.title}</Link>
  </li>
);

export default NavigationItem;
