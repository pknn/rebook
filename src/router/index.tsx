import React from 'react';
import { createBrowserHistory } from 'history';
import {
  connectRouter,
  routerMiddleware as getRouterMiddleware,
  ConnectedRouter as ConnectedRouterComponent
} from 'connected-react-router';
import { Switch, Route, Redirect } from 'react-router-dom';

import routesMap from './routesMap';

export const browserHistory = createBrowserHistory();
export const routerReducer = connectRouter(browserHistory);
export const routerMiddleware = getRouterMiddleware(browserHistory);
export const ConnectedRouter: React.FC = ({ children }) => (
  <ConnectedRouterComponent history={browserHistory}>
    {children}
  </ConnectedRouterComponent>
);

export const Routes: React.FunctionComponent = () => (
  <Switch>
    {routesMap.map((route) => (
      <Route key={route.path} path={route.path} exact={route.exact}>
        <route.component />
      </Route>
    ))}
    <Route path="*">
      <Redirect to="/not-found" />
    </Route>
  </Switch>
);
