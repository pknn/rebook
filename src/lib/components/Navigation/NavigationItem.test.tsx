import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Route } from 'router/routesMap';

import NavigationItem from './NavigationItem';

const mockRoute = (title: string, path: string): Route => ({
  title,
  path,
  component: jest.fn().mockReturnValue(<></>)
});

describe('<NavigationItem />', () => {
  describe('when it is the current route', () => {
    it('should render correctly', () => {
      const component = shallow(
        <NavigationItem route={mockRoute('mockTitle', 'mockPath')} isCurrent />
      );

      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });
  describe('when it is not the current route', () => {
    it('should render correctly', () => {
      const component = shallow(
        <NavigationItem
          route={mockRoute('mockTitle', 'mockPath')}
          isCurrent={false}
        />
      );

      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });
});
