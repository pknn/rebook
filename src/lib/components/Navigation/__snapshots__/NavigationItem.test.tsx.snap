// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> when it is not the current route should render correctly 1`] = `
<li
  className="mx-2 text-base text-gray-600 hover:text-yellow-500"
>
  <Link
    to="mockPath"
  >
    mockTitle
  </Link>
</li>
`;

exports[`<NavigationItem /> when it is the current route should render correctly 1`] = `
<li
  className="mx-2 text-base text-yellow-500"
>
  <Link
    to="mockPath"
  >
    mockTitle
  </Link>
</li>
`;
